components/molecules/AuthContentWrapper/auth-content-wrapper.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import clsx from "clsx";
+
+import Button from "components/atoms/Button/button";
+import Icon from "components/atoms/Icon/icon";
+
+import GitHubIcon from "img/icons/github-icon.svg";
+
+import useSupabaseAuth from "lib/hooks/useSupabaseAuth";
+
+interface AuthContentWrapperProps {
+  children: React.ReactNode;
+}
+
+const AuthContentWrapper: React.FC<AuthContentWrapperProps> = ({ children }) => {
+  const router = useRouter();
+  const currentPath = router.asPath;
+
+  const { signIn, user: isAuthed } = useSupabaseAuth();
+
+  const [host, setHost] = useState<string>("");
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      setHost(window.location.origin as string);
+    }
+  }, []);
+
+  return (
+    <div className="relative">
+      {!isAuthed && (
+        <div className="absolute z-10 w-full h-full top-0 left-0 flex items-center justify-center rounded-lg bg-white/50">
```

Comment:
```suggestion
        <div className="absolute z-10 w-full h-full flex items-center justify-center rounded-lg bg-white/50">
```
Do we need these classes??

Reply:
Yes, they are imperative to set the location of the absolutely positioned element.

Without them, the element isn't positioned properly to the top and left of its relative parent element.

Reply:
@babblebey IMHO, I think you set the display at your division `flex` and you use justify-center and items-center.

Reply:
Very well.

But `flex` and it's related properties `items-center` & `justify-center` are `display` properties for the element's child items i.e. the `Button` component, they determine how the component is rendered in box-types.

We require positioning for the element itself, hence the `absolute`, this allows us set "how" and "where" the element is positioned on the entire web page. 

The "how" is the `absolute` while the "where" are the `top` and `left. They work exactly together to create a crisp positioning.

In the absence of "where", we're living the location for the browser to decide, and this in most cases doesn't position the element well.